gymnasium/wrappers/transform_action.py

```diff
@@ -178,3 +179,107 @@ def __init__(
             func=func,
             action_space=act_space,
         )
+
+
+class DiscretizeAction(
+    TransformAction[ObsType, WrapperActType, ActType],
+    gym.utils.RecordConstructorArgs,
+):
+    """Discretizes a continuous Box action space into a single Discrete space.
+
+    Example 1 - Discretize Pendulum action space:
+        >>> env = gym.make("Pendulum-v1")
+        >>> env.action_space
+        Box(-2.0, 2.0, (1,), float32)
+        >>> obs, _ = env.reset(seed=42)
+        >>> obs, *_ = env.step([-0.6])
+        >>> obs
+        array([-0.17606162,  0.9843792 ,  0.5292768 ], dtype=float32)
+        >>> env = DiscretizeAction(env, bins=10)
+        >>> env.action_space
+        Discrete(10)
+        >>> obs, _ = env.reset(seed=42)
+        >>> obs, *_ = env.step(3)
+        >>> obs
+        array([-0.17606162,  0.9843792 ,  0.5292768 ], dtype=float32)
+
+    Example 2 - Discretize Reacher action space:
+        >>> env = gym.make("Reacher-v5")
+        >>> env.action_space
+        Box(-1.0, 1.0, (2,), float32)
+        >>> obs, _ = env.reset(seed=42)
+        >>> obs, *_ = env.step([-0.3, -0.5])
+        >>> obs
+        array([ 0.99908342,  0.99948506,  0.04280567, -0.03208766,  0.10445588,
+        0.11442572, -1.18958125, -1.97979484,  0.1054461 , -0.10896341])
+        >>> env = DiscretizeAction(env, bins=10)
+        >>> env.action_space
+        Discrete(100)
+        >>> obs, _ = env.reset(seed=42)
+        >>> obs, *_ = env.step(32)
+        >>> obs
+        array([ 0.99908342,  0.99948506,  0.04280567, -0.03208766,  0.10445588,
+        0.11442572, -1.1895813 , -1.97979484,  0.1054461 , -0.10896341])
+    """
+
+    def __init__(
+        self,
+        env: gym.Env[ObsType, ActType],
+        bins: int | tuple[int, ...],
+    ):
+        """Constructor for the discretize action wrapper.
+
+        Args:
+            env: The environment to wrap.
+            bins: int or list of ints (number of bins per dimension).
+        """
+        if not isinstance(env.action_space, Box):
+            raise TypeError(
+                "DiscretizeAction is only compatible with Box continuous actions."
+            )
+
+        self.low = env.action_space.low
+        self.high = env.action_space.high
+        self.n_dims = self.low.shape[0]
+
+        if not (np.all(np.isfinite(self.low)) and np.all(np.isfinite(self.high))):
```

Comment:
Shouldn't this be np.any(...) or np.any(...) as we can't allow any low or high to be infinite?

Reply:
`not np.all(np.isfinite(...))` is equivalent to what you said. I'll switch to yours because it is clearer.